runtime/sema/type_tags.go

```diff
@@ -650,6 +655,9 @@ func findSuperTypeFromUpperMask(joinedTypeTag TypeTag, types []Type) Type {
 	case invalidTypeMask:
 		return InvalidType
 
+	case interfaceTypeMask:
+		return getSuperTypeOfDerivedTypes(types)
+
```

Comment:
We can add `interfaceTypeMask` to the case below, which also calls the `getSuperTypeOfDerivedTypes` method. No need to add a new case.

Reply:
Done.